### DISCUSSION TITLE
Dynamic Chart based on table filtering

### DISCUSSION THREAD

**Discussion by tomerikko** [EvotecIT/PSWriteHTML#390] [Q&A]
Hey All, i wonder if this is possible.

I am looking to build a chart (pie\bars or similar) that is connected to a table that will resides next to her.
what i want to happen is that the chart will represent the summary of the table, and if i filter the table the chat data will update automatically.

Does this make sense?
Any chance this is already possible and someone could share some advice?

Love this tool BTW - thanks for both creating and maintaining it.

**Comment by PrzemyslawKlys** [comment]
It is possible - https://evotec.xyz/nested-tabs-diagram-updates-diagram-events-calendar-object-and-more-in-pswritehtml/ 

Edit: hrmms, its not in this blog. But i did it for sure. Should be somewhere in examples

**Reply by PrzemyslawKlys** [reply]
```powershell
Import-Module .\PSWriteHTML.psd1 -Force

New-HTML {
    New-HTMLPanel {
        $DataTable = @(
            [PSCustomObject] @{
                Name  = 'My Object 1'
                Time  = 1
                Money = 5
                Taxes = 20
            }
            [PSCustomObject] @{
                Name  = 'My Object 2'
                Time  = 3
                Money = 1
                Taxes = 5
            }
            [PSCustomObject] @{
                Name  = 'My Object 3'
                Time  = 12
                Money = 5
                Taxes = 1
            }
            [PSCustomObject] @{
                Name  = 'My Object 4'
                Time  = 15
                Money = 20
                Taxes = 30
            }
        )

        New-HTMLTable -DataTable $DataTable -DataTableID 'NewIDtoSearchInChart' -Filtering
        New-HTMLChart {
            New-ChartToolbar -Download
            New-ChartLegend -Name 'Time', 'Money', 'Taxes'
            foreach ($Object in $DataTable) {
                New-ChartBar -Name $Object.Name -Value $Object.Time, $Object.Money, $Object.Taxes
            }
            New-ChartEvent -DataTableID 'NewIDtoSearchInChart' -ColumnID 0
        }
        <#
        New-HTMLTable -DataTable $DataTable -DataTableID 'Opps' -Filtering
        New-HTMLChart {
            New-ChartToolbar -Download
            New-ChartLegend -Name 'Time', 'Money', 'Taxes'
            foreach ($Object in $DataTable) {
                New-ChartBar -Name $Object.Name -Value $Object.Time, $Object.Money, $Object.Taxes
            }
            New-ChartEvent -DataTableID 'Opps' -ColumnID 0
        }
        #>
    }
} -ShowHTML -FilePath $PSScriptRoot\Example-ChartsWithTablesBar.html -Online
```

Click on chart - you get table filtered. I am not sure it can be done the other way around at the moment.

**Reply by tomerikko** [reply]
hey @PrzemyslawKlys   , thanks for sharing.

I actually meant the opposite.
So think you have a table that looks like this:

Team           Region          Items_Sold       Items_In_Stock       Items_Ordered
Team A       EMEA                  10                         10                       5  
Team A       APAC                   2                           20                       3  
Team A      AMER                   5                           10                       15  
Team B       EMEA                   7                            3                         3  
Team B       APAC                   22                         44                       33  
Team B      AMER                   11                         14                       11  
Total          All                        57                        101                      70

so if i would create a pie chart on this data, i would go with the TOTAL, to present 3 portions, the numebrs of the totla: Items_Sold       Items_In_Stock       Items_Ordered   which is:  57 / 101 / 70


then if i want to know only how Team A is doing around the globe, i would filter the Talbe on the team, now my table is:
Team           Region          Items_Sold       Items_In_Stock       Items_Ordered
Team A       EMEA                  10                         10                       5  
Team A       APAC                   2                           20                       3  
Team A      AMER                   5                           10                       15  
Total          All                        17                        40                        23

so the pie chart would be with values:   17  /  40  /  23


then i might change my mind and what to see how EMEA is doing, so i re-fitler the table to only show emea, now my table is:
Team           Region          Items_Sold       Items_In_Stock       Items_Ordered
Team A       EMEA                  10                         10                       5  
Team B       EMEA                   7                            3                        3  
Total          All                        17                        13                        8
so now the pie chart would be with values:   17  /  13  /  8


*if the total is presented or calculated in the backend, is less important to me ATM.




do i make any sense?

**Reply by PrzemyslawKlys** [reply]
You do, but you need to think on how you connect this data with chart. The way you described it - you basically would need to have 3 separate charts. The charts should be hidden, until you actually click on some row in Table, and then show the data. I don't see a way to filter anything here because the first chart, doesn't contain data for 2nd and 3rd one. So what you essentially need is total rebuild on search/filtering.

**Reply by tomerikko** [reply]
Make sense thanks for the insights.
My idea was to "inject" the data that is lively displayed on the table into the chart, and therefore the chart data will get changed\refreshed on each change in the data presented on the table.


kind of piping the table displayed content into the chart.


This might not be possible today, and its fine!    just mentioning a way to interact with the data .
what i do right now is similar to your suggestion,   i have 3 charts 1 for each way to filter the data, but tbh, it takes a lot of "real estate"  and TBH, to add everything i want, i need to add 20 more :)
so if i can somewhere in the future, have 1 that is directly connected to a table and it displayed only what the table display, it would be awesome!!!!!!!!!!!!!!!!!!!!!!!! :)
